pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -711,3 +713,80 @@ func (r *reconcilerBase) isWebhookEnabled(ctx context.Context) (bool, error) {
 	}
 	return err == nil, err
 }
+
+// patchSyncMetadata patches RSync metadata:
+// - Set label "applyset.kubernetes.io/id" (see applyset.IDFromSync for value format)
+// - Set annotation "applyset.kubernetes.io/tooling" (see applyset.FormatTooling for value format)
+//
+// This method uses a merge-patch, instead of server-side apply, because the
+// reconciler-manager does not own the RSync. No other fields should be modified.
+//
+// Returns true if changes were made, false if no changes were necessary.
+//
+// Returns an error if the tooling annotation exists with a different name or
+// an invalid value.
+func (r *reconcilerBase) patchSyncMetadata(ctx context.Context, rs client.Object) (bool, error) {
+	syncScope := declared.ScopeFromSyncNamespace(rs.GetNamespace())
+	applySetID := applyset.IDFromSync(rs.GetName(), syncScope)
+	applySetToolingValue := applyset.FormatTooling(metadata.ApplySetToolingName, metadata.ApplySetToolingVersion)
+	updateRequired := false
+
+	currentApplySetID := core.GetLabel(rs, metadata.ApplySetParentIDLabel)
+	if currentApplySetID != applySetID {
+		updateRequired = true
+
+		if r.logger(ctx).V(5).Enabled() {
+			r.logger(ctx).Info("Updating sync metadata: applyset label",
+				logFieldResourceVersion, rs.GetResourceVersion(),
+				"labelKey", metadata.ApplySetParentIDLabel,
+				"labelValueOld", currentApplySetID, "labelValue", applySetID)
+		}
+	}
+
+	currentApplySetToolingValue := core.GetAnnotation(rs, metadata.ApplySetToolingAnnotation)
+	if currentApplySetToolingValue != applySetToolingValue {
+		updateRequired = true
+
+		// If already set, validate that we can safely adopt the ApplySet
+		if currentApplySetToolingValue != "" {
```

Comment:
When would this happen? If we bump the `metadata.ApplySetToolingVersion`?

Reply:
Yeah. I figured I'd add it so old versions wouldn't freak out if we decided to change it in the future and someone had to downgrade.

Reply:
I assume we would need to flesh out the user journey more in that case, right? I figure we don't expect the user to manually go in and mutate the old annotations

Reply:
1. If you upgrade and we've changed the version, it would auto-upgrade all the RSyncs to have the new version.
2. If you downgrade after we've changed the version, it would auto-downgrade all the RSyncs to have the new version.
3. We don't allow multiple versions of Config Sync installed at the same time, so we shouldn't have to deal with any fights or conflicts.
4. If we DID allow multiple versions concurrently, we'd need to rethink how this behavior works to make sure there was some way to delineate which reconciler-manager managed which RSyncs and how to manage upgrades/downgrades without losing access to the RSyncs we owned before. This would probably require a new label with a unique name for each install.